syntax = "proto3";

package proto;

message KeyEvent {
    bool down = 1;
    int32 key = 2;
}

message VsplitProto {
    int32 id = 1;
    bytes vsplit = 2;
}

message PMeshProto {
    int32 id = 1;
    int32 nvertices = 2;
    bytes pmesh_info = 3;
    bytes base_mesh = 4;
    repeated VsplitProto vsplit = 5;
}

message Frame3DProto {
    bytes view_matrix = 1;
    bytes color = 2;
    bytes depth = 3;
}

message CommProto {
    int32 mouse_x = 1;
    int32 mouse_y = 2;
    repeated KeyEvent key_event = 3;
    Frame3DProto diff_frame = 4;
    repeated PMeshProto pmesh = 5;
    bool disconnect = 6;
}
